.github/workflows/macos-compile-test.yml

```diff
@@ -0,0 +1,76 @@
+name: macOS Compile EAS
+
+on:
+  workflow_dispatch: {}
+  pull_request:
+    paths:
+      - .github/workflows/macosos-compile-test
+      - packages/expo-modules-core/e2e/setup/create-eas-project-macos.ts
+  push:
+    branches: [main, 'sdk-*']
```

Comment:
do we need to run the test on sdk-* branch whenever pushes to `packages/expo/**` or some else packages?